ALL : Keep adaptive links only

BellezaEmporium opened the pull request:
We talked about this for some time, to prevent using chunklists, we prefer keeping adaptive links (playlist.m3u8, index.m3u8, master.m3u8 etc etc...) that can permit to get the best quality for your bandwidth (depending on what your bandwidth can take, you'll get the equivalent quality to your bandwidth.)

This PR will try to change all the links to keep only playlists.

PS : If I find that the playlist only contains one single line to a chunklist, then I keep the chunklist instead of a playlist that redirects to the same chunklist.

BellezaEmporium committed:
ALL : Chunklist replacements

We talked about this for some time, to prevent using chunklists, we prefer keeping adaptive links (playlist.m3u8, index.m3u8, master.m3u8 etc etc...) that can permit to get the best quality for your bandwidth (depending on what your bandwidth can take, you'll get the equivalent quality to your bandwidth.)

This PR will try to change all the links to keep only playlists.

freearhey commented:
I suggest converting PR to "Draft" to let everyone know that it is not yet ready.

BellezaEmporium committed:
AE : Cleanup, fix IDs and delete chunklists

BellezaEmporium committed:
AL : Delete chunklists

BellezaEmporium committed:
AM : Replace links and delete chunklists

Dum4G commented on channels/ae.m3u:
```diff
@@ -233,8 +159,6 @@ https://weyyak-live.akamaized.net/weyyak_drama/index.m3u8
 #EXTINF:-1 tvg-id="WeyyakNawaem.ae" tvg-name="Weyyak Nawaem" tvg-country="AE" tvg-language="Arabic" tvg-logo="https://content.weyyak.com/d867c2e9-328d-ea11-831d-02d662556796/poster-image" group-title="",Weyyak Nawaem (720p)
 https://weyyak-live.akamaized.net/weyyak_nawaem/index.m3u8
-#EXTINF:-1 tvg-id="Yas.ae" tvg-name="Yas" tvg-country="AE" tvg-language="" tvg-logo="" group-title="",Yas (1080p)
-https://admdn1.cdn.mangomolo.com/yastv/smil:yastv.stream.smil/playlist.m3u8?adtv
```
Why, it works

LaneSh4d0w replied:
Duplicate of another one.

LaneSh4d0w replied:
> 
> 
> Why? that works. Overall, I see some channels being removed without a replacement. That's not our way to manage things

They were basically the same channels, but only a different chunklist. All of them work, but we keep only one adaptive link that simply do the same thing as keeping 5 or 6 chunk links.

Dum4G replied:
I see only one Yas here and it shows Yas icon on the channel

LaneSh4d0w replied:
Line 163.

Dum4G replied:
Also, we should retain all logos, groups etc if at least a single entry had them

LaneSh4d0w replied:
That's what I try to do.

BellezaEmporium committed:
AR : Delete chunklists and replace links

Dum4G reviewed (changes requested):
Why? that works. Overall, I see some channels being removed without a replacement. That's not our way to manage things

BellezaEmporium committed:
AD : Delete dupe link

Dum4G commented:
You can safely keep links that contain just "chunklist.m3u", sometimes it is the same thing as "playlist.m3u"
Ekta is a good example if you shorten the link

LaneSh4d0w commented:
> 
> 
> You can safely keep links that contain just "chunklist.m3u", sometimes it is the same thing as "playlist.m3u"
> Ekta is a good example if you shorten the link

Ekta was a wrongly named Dubai One, if I remember well.

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
AT : Delete chunklists

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
AU : Delete chunklists when possible and replace links

BellezaEmporium committed:
AW : Delete dupes and chunklists

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
AZ : Replaced links

BellezaEmporium committed:
BA : Replaced links (couldn't test them all)

BellezaEmporium committed:
BB : Delete dupe link (in playlist)

BellezaEmporium committed:
BE : Replace links when needed / Deleted dupes

Dum4G commented:
RTRS should go live in the next hour

LaneSh4d0w commented:
> RTRS should go live in the next hour

Thanks for telling me. I'll check if the links in playlist are litterally the ones that exist.

Dum4G commented:
I can confirm the website is offering me this link but I have CORS error while accessing it. I suppose it's geoblocked
https://rtrs.tv/program/uzivo.php?c=1&bcast_on=1

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

LaneSh4d0w commented:
> 
> 
> I can confirm the website is offering me this link but I have CORS error while accessing it. I suppose it's geoblocked
> https://rtrs.tv/program/uzivo.php?c=1&bcast_on=1

Same for me, i'll try using a proxy to see if it works.

LaneSh4d0w commented:
I agree, it's geo-blocked to Bulgaria.

BellezaEmporium committed:
BA : Change link to RTRS and add geo-lock mention.

BellezaEmporium committed:
BF : Change link for Impact TV

BellezaEmporium committed:
BG : Change one link (couldn't verify all of them)

BellezaEmporium committed:
BH : Deleted dupes / Changed chunklists for playlists

BellezaEmporium committed:
BN : Changed chunklist for playlist

BellezaEmporium committed:
BO : Deleted dupes

BellezaEmporium committed:
BR : Changes were made

BellezaEmporium committed:
Update br.m3u

BellezaEmporium committed:
BS : Delete chunklist link

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
AE : Delete chunklists/dupes, reorganized playlist

BellezaEmporium committed:
AF : Delete chunklists and replace stuff

BellezaEmporium committed:
Update af.m3u

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

BellezaEmporium committed:
US : Begin cleaning

This will be a pain in the *** to clean everything, but we'll try our best.

Beginning with the 3ABN channels.

BellezaEmporium committed:
Merge branch 'LaneSh4d0w-patch-1' of https://github.com/iptv-org/iptv into LaneSh4d0w-patch-1

Dum4G committed:
Merge branch 'master' into LaneSh4d0w-patch-1

Dum4G commented:
I don't think this job should be done by hand. Let me play around with REGEXP a bit and gather as much cases of non-adaptive links as possible. If everything will be performed right we'll have hundred of duplicate links that will be removed the same day automatically. You can merge this the way it is I guess

LaneSh4d0w commented:
> 
> 
> I don't think this job should be done by hand. Let me play around with REGEXP a bit and gather as much cases of non-adaptive links as possible. If everything will be performed right we'll have hundred of duplicate links that will be removed the same day automatically. You can merge this the way it is I guess

I suppose... You should also verify that links are working at the same instance. There might be instances in which "master.m3u8" only works, instances where only "index.m3u8" works, or even instances where only "playlist.m3u8" works.

Dum4G commented:
That's why we don't auto-merge things. I will try to handle this based on a source where the link comes from

LaneSh4d0w commented:
This is a first set of many more PRs to come.

BellezaEmporium committed:
Merge branch 'master' into LaneSh4d0w-patch-1

Dum4G committed:
fix typo

Dum4G commented on channels/bo.m3u:
```diff
@@ -9,29 +9,13 @@ https://mediacp.hostradios.com.ar:19360/atbcochabamba/atbcochabamba.m3u8
 #EXTINF:-1 tvg-id="BoliviaTV72.bo" tvg-name="Bolivia TV 7.2" tvg-country="BO" tvg-language="Spanish" tvg-logo="https://i.imgur.com/1zTLGdj.jpg" group-title="",Bolivia TV 7.2 (720p)
 https://video1.getstreamhosting.com:1936/8224/8224/chunklist_w1684483949.m3u8
-#EXTINF:-1 tvg-id="BoliviaTV72.bo" tvg-name="Bolivia TV 7.2" tvg-country="BO" tvg-language="Spanish" tvg-logo="https://i.imgur.com/1zTLGdj.jpg" group-title="",Bolivia TV 7.2 (720p)
-https://video1.getstreamhosting.com:1936/8224/8224/playlist.m3u8
```
works

LaneSh4d0w replied:
It's because the playlist only has got the chunklist above, not useful to keep it.

Dum4G commented on channels/br.m3u:
```diff
@@ -2,18 +2,12 @@
 #EXTINF:-1 tvg-id="Animestation.br" tvg-name="Animestation" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/5UpjGcL.png" group-title="",Animestation (480p) [Not 24/7]
 https://stmv.video.expressolider.com.br/animestation1/animestation1/chunklist_w2074350440.m3u8
-#EXTINF:-1 tvg-id="Animestation.br" tvg-name="Animestation" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/5UpjGcL.png" group-title="",Animestation (480p) [Not 24/7]
-https://stmv.video.expressolider.com.br/animestation1/animestation1/playlist.m3u8
```
works

LaneSh4d0w replied:
It's because the playlist only has got the chunklist above, not useful to keep it.

Dum4G commented on channels/br.m3u:
```diff
@@ -101,28 +87,24 @@ https://v4-slbps-sambavideos.akamaized.net/live/3282%2C8114%2Cec4b5a296d97fa99bf
 https://hls.brasilstream.com.br/live/redetves/redetves/tracks-v1a1/mono.m3u8
-#EXTINF:-1 tvg-id="RedeVida.br" tvg-name="Rede Vida" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/rJ0kO2o.png" group-title="",Rede Vida (720p)
-https://cvd1.cds.ebtcvd.net/live-redevida/smil:redevida.smil/chunklist.m3u8
 #EXTINF:-1 tvg-id="Rede Vida" tvg-name="Rede Vida" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/h4WaIw5.jpg" group-title="Religious",Rede Vida (720p)
```
bad tvg id

LaneSh4d0w replied:
Will need to fix it

Dum4G commented on channels/br.m3u:
```diff
@@ -101,28 +87,24 @@ https://v4-slbps-sambavideos.akamaized.net/live/3282%2C8114%2Cec4b5a296d97fa99bf
 #EXTINF:-1 tvg-id="RedeSeculo21.br" tvg-name="Rede Século 21" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/vvrCWVo.jpg" group-title="Religious",Rede Século 21 (240p) [Not 24/7]
 https://d1gs793b1l6k9i.cloudfront.net/out/v1/a3aac39375df40a1a564635d7b41e90c/index.m3u8
 #EXTINF:-1 tvg-id="RedeTVES.br" tvg-name="Rede TV! ES" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://www.livetvnews.com.br/wp-content/uploads/2019/03/Logo-RedeTV.png" group-title="General",Rede TV! ES
 https://hls.brasilstream.com.br/live/redetves/redetves/tracks-v1a1/mono.m3u8
```
redundant tracks-v1a1

LaneSh4d0w replied:
It's because the playlist file only has got this version, not useful to keep it.

Dum4G commented on channels/br.m3u:
```diff
@@ -286,19 +264,15 @@ http://flash.softhost.com.br:1935/ufg/tvufgweb/playlist.m3u8
 #EXTINF:-1 tvg-id="TVCIParanagua.br" tvg-name="TVCI Paranaguá" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/4GKZbXD.png" group-title="General",TVCI Paranaguá (360p)
 http://flash8.crossdigital.com.br/2306/2306/playlist.m3u8
-#EXTINF:-1 tvg-id="TVE.br" tvg-name="TVE" tvg-country="BR" tvg-language="" tvg-logo="" group-title="",TVE
-http://stream2.ba.gov.br:80/hls-live/livepkgr/_definst_/irdeb/pgm-1.m3u8
```
works

LaneSh4d0w replied:
I think I had changed this version. Don't quite remember for this one.

Dum4G replied:
Any ideas?

LaneSh4d0w replied:
On retest it works. Maybe I was trying to search for a better version but went out.

Reinstating it.

BellezaEmporium committed:
Update br.m3u

Dum4G committed:
Update br.m3u

Dum4G committed:
Merge branch 'master' into LaneSh4d0w-patch-1
